### PR TITLE
feat: migrate and refactor replica-healthcheck from op-replica

### PR DESCRIPTION
**Pull request by annieke**
**Description**
We have been running this healthcheck server alongside replicas for a while, and this PR migrates and refactors our original code to bring it into the monorepo. 

**Additional context**
Original repository: https://github.com/optimisticben/op-replica 

**Metadata**
- Fixes OP-1019

### CODE REVIEW

File: packages/replica-healthcheck/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```
Comment (snario) [inline packages/replica-healthcheck/LICENSE:22]: Do we need another license? Wouldn't it inherit from the root directory?

File: packages/replica-healthcheck/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Here's a list of environment variables:
+
+| Variable                 | Purpose                                                  | Default                                                                                  |
+| ------------------------ | -------------------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| ETH_NETWORK              | Ethereum Layer1 and Layer2 network (mainnet,kovan)       | mainnet (change to `kovan` for the test network)                                         |
+| ETH_NETWORK_RPC_PROVIDER | Layer2 source of truth endpoint, used for the sync check | https://mainnet.optimism.io (change to `https://kovan.optimism.io` for the test network) |
+| ETH_REPLICA_RPC_PROVIDER | Layer2 local replica endpoint, used for the sync check   | http://localhost:9991                                                                    |
+| L2GETH_IMAGE_TAG         | L2geth version                                           | 0.4.                                                                                     |
```
Comment (snario) [inline packages/replica-healthcheck/README.md:37]: Could these be prefixed with `REPLICA_HEALTHCHECK__`?

File: packages/replica-healthcheck/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@eth-optimism/replica-healthcheck",
+  "version": "0.1.0",
+  "private": true,
+  "main": "dist/index",
+  "files": [
+    "dist/index"
```
Comment (snario) [inline packages/replica-healthcheck/package.json:7]: Are you sure this is the only file to be distributed?

File: packages/replica-healthcheck/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "@types/express": "^4.17.12",
+    "@types/node": "^15.12.2",
+    "ts-mocha": "^8.0.0",
+    "ts-node": "^10.0.0",
+    "typescript": "^4.3.2"
+  },
+  "dependencies": {
+    "@eth-optimism/common-ts": "0.1.5",
+    "@eth-optimism/core-utils": "^0.5.1",
+    "dotenv": "^10.0.0",
```
Comment (snario) [inline packages/replica-healthcheck/package.json:32]: Also a dev dependency, no?

File: packages/replica-healthcheck/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "ethers": "^5.3.0",
+    "express": "^4.17.1",
+    "express-prom-bundle": "^6.3.6",
+    "prom-client": "^13.1.0",
+    "supertest": "^6.1.4"
```
Comment (snario) [inline packages/replica-healthcheck/package.json:37]: dev dependency?

**Review by snario — APPROVED** [review]

File: packages/replica-healthcheck/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as dotenv from 'dotenv'
```
Comment (tynes) [inline packages/replica-healthcheck/src/index.ts:1]: `src/index` usually exports the package as a library and executables usually go in something like `bin/health-check-server` or `exec/health-check-server`

File: packages/replica-healthcheck/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+import { HealthcheckServer } from './healthcheck-server'
+import { readConfig } from './helpers'
+;(async () => {
+  dotenv.config()
+
+  const healthcheckServer = new HealthcheckServer(readConfig())
+
+  healthcheckServer.init()
+  healthcheckServer.runSyncCheck()
```
Comment (tynes) [inline packages/replica-healthcheck/src/index.ts:11]: Does this need an `await`?

File: packages/replica-healthcheck/package.json (hunk 3 above)
Comment (tynes) [reply packages/replica-healthcheck/package.json:7]: This is only useful for npm packages if I recall

**Comment by tynes** [comment]
Generally looks good, nice job

File: packages/replica-healthcheck/src/index.ts (hunk 7 above)
Comment (annieke) [reply packages/replica-healthcheck/src/index.ts:11]: the thought was that multiple checks could happen synchronously when more is added, does not adding an `await` here make sense in that case?

File: packages/replica-healthcheck/LICENSE (hunk 1 above)
Comment (annieke) [reply packages/replica-healthcheck/LICENSE:22]: every package has a copy of the license rn!

File: packages/replica-healthcheck/src/index.ts (hunk 7 above)
Comment (annieke) [reply packages/replica-healthcheck/src/index.ts:11]: just realized that without the await the app doesn't properly quit on errors (our desired behavior), so added!

**Pull request merged**